lib/screens/user/blood_donation_reservation_page.dart

```diff
@@ -62,29 +63,27 @@ class _BloodDonationReservationPageState
 
   void _submitForm() async {
     if (_formKey.currentState!.validate() && _selectedBloodType != null) {
-      await FirebaseFirestore.instance.collection('donation_date').add({
-        'donor_name': _nameController.text,
-        'email': _emailController.text,
-        'date': _dateController.text,
-        'blood_type': _selectedBloodType,
-        'user_id': _userId
-      });
+      await _donationService.saveBloodDonationData(
+        donorName: _nameController.text,
+        email: _emailController.text,
+        date: _dateController.text,
+        bloodType: _selectedBloodType!,
+        userId: _userId!,
+      );
 
-      // Clear form data and reset blood type dropdown
       _nameController.clear();
       _emailController.clear();
       _dateController.clear();
       setState(() {
         _selectedBloodType = null;
       });
 
-      // Show success dialog
       showDialog(
         context: context,
         builder: (context) => AlertDialog(
           title: const Text('Success'),
           content:
-              const Text('Your blood donation reservation has been submitted.'),
+          const Text('Your blood donation reservation has been submitted.'),
```

Comment:
just question:

Could text somehow be moved in some kind of csv file, or something similar?

So in case of localization it's easier to use placeholder values and different languages files?

Reply:
I'm currently working on localization in parallel with this.
It will be done as soon as possible